src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1770,6 +1771,81 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
             if (notification.HandledResponse) {
                 return;
             }
+            
+            var identifier = ticket.GetProperty(OpenIdConnectConstants.Extra.ClientId);
```

Comment:
> Ah. I assumed we would want implementers of the ValidationEndpoint to be able to modify the incoming data for the ValidationEndpointNotification if they needed to do some custom processing to change things if they didn't want to allow the default logic.

ASOS is not that permissive :smile: 

The different endpoint events are always invoked after the minimal checks, to avoid calling user code with potentially malformed/corrupted/unsafe data.

I'm convinced the security checks added by this PR shouldn't be skipped (at least, not easily: you're of course free to replace/redirect the whole endpoint logic directly in `MatchEndpoint`).